models/user.go

```diff
@@ -75,3 +76,48 @@ func (user User) Get(c *config.Conf) (User, error) {
 	}
 	return user, nil
 }
+
+// CheckUser checks if a user exists in the database
+func (user User) CheckUser(c *config.Conf) (User, error) {
+
+	mgoSession := c.Database.Session.Copy()
+	defer mgoSession.Close()
+
+	collection := c.Database.C(config.USERCOLLECTION).With(mgoSession)
+
+	result := User{}
+	// log.Println("user.Email: ", user.Email)
+
+	err := collection.Find(bson.M{"email": user.Email}).One(&result)
+	if err != nil {
+		log.Println("User not found:", err)
+		// log.Println("error User: ", user)
+
+		// return result, errors.New("Username or password is incorrect (no user)")
+		return result, err
+	}
+
+	// log.Println("USER: ", user, "result:", result)
+	// if result.Name == "" {
+	// 	return result, errors.New("Username or Password is incorrect(no user)")
+	// }
+
+	return result, nil
+}
+
+// Insert : inserts user data to the DB
+func (user User) Insert(c *config.Conf) error {
```

Comment:
like i said in the message, what could be done is to rewrite the upsert function to insert new content if it doesnt exist, or update the existing content with the new data. ie if user exists, and email doesnt exist, add email. basically update the fields that have changed. It might be possible using $set in upsert. but you could research it, or even use multiple database calls if necessary. to request the data and compare changes. If you just add data without checks, whether with new UUID or not, you could have two accounts pointing to one user. one from email signup and the other from email